src/core/core.js

```diff
@@ -48,13 +32,19 @@ function type(obj) {
         return 'null';
     }
 
-    const type = typeof obj;
-
-    if (type === 'undefined' || type === 'number' || type === 'string' || type === 'boolean') {
-        return type;
+    const typeString = typeof obj;
+    if (['undefined', 'number', 'string', 'boolean'].includes(typeString)) {
```

Comment:
this array should not be constructed each time the function runs but be global

Reply:
I tested your theory because I don't believe static arrays or objects are constructed each time the function is called. (Just like a number 52.3 or "string" appearing in a function isn't "constructed" each time). Only a stupid JS implementation would do that...

In Chrome this is true: no extra cpu or memory overhead using global or locally declared static array or object.

However in safari the locally declared version runs 25% slower.

So I'll move it.

Reply:
Interesting, good to know!